### DISCUSSION TITLE
I want to contribute to PyO3 and would like suggestions where I can help!

### DISCUSSION THREAD

**Discussion by davidhewitt** [PyO3/pyo3#1920] [General]
Thank you for your interest in contributing to PyO3! Please feel free to post in this discussion with your experience and what kind of stuff you're interested in working on and I'll try to share ideas with you on what might suit.

Some general suggestions that might be relevant to anyone:
 - I've labelled some issues with [Good First Issue](https://github.com/PyO3/pyo3/issues?q=is%3Aopen+is%3Aissue+label%3A%22Good+First+Issue%22). Some of these might be small and easy; they might be well-defined problems to solve. Often they are expected to only need to work on a specific part of the codebase, and don't have too broad design questions still waiting.
 - The [milestones](https://github.com/PyO3/pyo3/milestones) contain a bunch of issues which I'd hope to see implemented in the upcoming releases because I think we've nearly solved all the technical challenges on the path to solving them.
 - [Documentation issues](https://github.com/PyO3/pyo3/issues?q=is%3Aopen+is%3Aissue+label%3Adocumentation) are a brilliant way to familiarise with the codebase at the same time as helping future users out. We have made huge progress with our docs recently, however there's no upper limit to how good the documentation and guide can be and it greatly improves the experience for first-time Rust users coming from Python. In particular, we really want:
   - Documentation for everything except the ffi module.
   - Every item should have a doc example(s).
 - Review open PRs - a great way to learn about current projects we're working on in bitesize chunks. Any and all comments and questions on PRs welcome, no matter how basic they may be.
 - Feel free to just read the code and open PRs for refactorings / optimizations if you spot anything!
 - Help with design and implementation of tickets proposed for the [1.0 Release](https://github.com/PyO3/pyo3/projects/1)

**Comment by davidhewitt** [comment]
@aganders3 - continuing your question from https://github.com/PyO3/pyo3/issues/1820#issuecomment-942326525, here's a few ideas where you could look for ways to get involved in this project.

Unfortunately a lot of the things I've labelled "Good First Issue" do still require changes to proc macro code, however I'm always happy to offer mentoring to anyone who wants to help out, so let me know if any of the suggested things take your fancy!

**Reply by aganders3** [reply]
Thanks! I will take a look. I know it takes effort for existing maintainers to make it easy to contribute, so I appreciate it. Hopefully in time I can add some value to the project.

**Reply by jovenlin0527** [reply]
https://github.com/dtolnay/proc-macro-workshop is wonderful for learning how to implement proc macro.

**Reply by aganders3** [reply]
> https://github.com/dtolnay/proc-macro-workshop is wonderful for learning how to implement proc macro.

Thanks! I'm definitely adding this to my reading list.

**Comment by aviramha** [comment]
@davidhewitt I'd add working on tickets based on https://github.com/PyO3/pyo3/projects/1, more importantly 1.0-must so we can push the first stable version ;)

**Reply by davidhewitt** [reply]
Added! The thorny thing with 1.0-must is that the implementation of those is likely very hard and requires knowledge of most of PyO3. Design opinions to help with end-user outcomes is very very welcome for 1.0-must, because there's a lot (especially for #1056) which still needs decisions.

1.0-candidate issues are good for most contributors to angle for!
